File: couch/csharp/read/couch_read.cs
// ------------------------------------------------------------------
//	read/couch_read.cs
//
//					May/07/2015
//
// ------------------------------------------------------------------
using System;
using System.IO;
using System.Net;
using System.Text;
using System.Collections.Generic;

using Newtonsoft.Json;

// ------------------------------------------------------------------
public class couch_read
{
// ------------------------------------------------------------------
static void Main (string[] args)
{
	Console.WriteLine ("*** 開始 ***");

	string url = "http://localhost:5984/nagano";

	string user = "";
	string password = "";

	string url_all_docs = url + "/_all_docs?include_docs=true";

	string str_json = get_uri.get_uri_proc (url_all_docs,user,password);

	Dictionary<String,Object> all_docs
		= JsonConvert.DeserializeObject <Dictionary<String,Object>> (str_json);

	String bbb_json = all_docs["rows"].ToString ();
	List <Dictionary<String,Object>> bbb_pp
	= JsonConvert.DeserializeObject <List <Dictionary<String,Object>>>
		 (bbb_json);

	for (int it=0; it< bbb_pp.Count; it++)
		{
		string key = bbb_pp[it]["key"].ToString ();

		string str_unit = bbb_pp[it]["doc"].ToString ();
		kvalue_manipulate.out_record_proc (key,str_unit);
		}

	Console.WriteLine ("*** 終了 ***");
}

// ------------------------------------------------------------------
}
// ------------------------------------------------------------------

File: couch/fsharp/read/couch_read.fs
// -------------------------------------------------------------------
//
//	couch_read.fs
//
//					Mar/03/2016
//
// -------------------------------------------------------------------
open System
open System.Text
open Microsoft.FSharp.Core
open System.IO
open System.Net
open System.Collections
open System.Collections.Generic

open Newtonsoft.Json
// -------------------------------------------------------------------
let list_parser (list_pp: List<Dictionary<String,Object>>) =
  for unit in list_pp do
  let key = unit.Item ("key")
  let doc = unit.Item ("doc")
  let str_json_out = JsonConvert.SerializeObject (doc)
  let dict_cc = JsonConvert.DeserializeObject<Dictionary<String,Object>> (str_json_out)
  let name = dict_cc.Item ("name")
  let population = dict_cc.Item ("population")
  let date_mod = dict_cc.Item ("date_mod")
  Console.Write (key.ToString () + "\t")
  Console.Write (name.ToString () + "\t")
  Console.Write (population.ToString () + "\t")
  Console.Write (date_mod.ToString () + "\n")
  done
// -------------------------------------------------------------------
[<EntryPoint>]
let main (args : string[] ) =

  printfn "*** 開始 ***"

  let url = "http://localhost:5984/nagano"

  let user = ""
  let password = ""

  let url_all_docs = url + "/_all_docs?include_docs=true"
  let str_json = get_uri.get_uri_proc (url_all_docs,user,password)

  let all_docs = JsonConvert.DeserializeObject<Dictionary<string,Object>>(str_json)

//  Console.WriteLine (all_docs.["total_rows"].ToString ())
  let bbb_json = all_docs.["rows"].ToString ()
  let list_pp = JsonConvert.DeserializeObject<List<Dictionary<String,Object>>> (bbb_json)
//  Console.WriteLine (list_pp.Count)

  let _ = list_parser (list_pp)

  printfn "*** 終了 ***"

  0

// ------------------------------------------------------------------
